### PR TITLE
Refactor SoapyCake

### PR DESCRIPTION
**Pull request by dziemba**

### CODE REVIEW

File: soapy_cake.gemspec [hunk 1]
```diff
@@ -14,13 +14,13 @@ Gem::Specification.new do |spec|
   spec.license       = 'MIT'
 
   spec.files         = `git ls-files -z`.split("\x0")
-  spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
-  spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
+  spec.executables   = spec.files.grep(/^bin\//) { |f| File.basename(f) }
```
Comment (sncf) [inline soapy_cake.gemspec]: Why `/` instead of `%r{`?

Comment (dziemba) [inline soapy_cake.gemspec]: rubocop said so ;)

File: lib/soapy_cake/request.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    def initialize(role, service, method, opts = {})
+      @role = role.to_s
+      @service = service.to_s
+      @method = method.to_s
+      @opts = opts
+    end
+
+    def path
+      "/#{(role != 'admin') ? role : ''}/api/#{version}/#{service}.asmx"
```
Comment (sncf) [inline lib/soapy_cake/request.rb]: `#{role if role != 'admin'}`, but it's OK.

File: soapy_cake.gemspec (hunk 1 above)
Comment (sncf) [inline soapy_cake.gemspec]: Strange. Usually it complains if I use `//` syntax with a regexp containing slashes.

**Pull request merged**